File: 0905-sort-array-by-parity/0905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int n = nums.size();
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                if(nums[i]%2 !=0 && nums[j]%2 == 0){
                    swap(nums[i],nums[j]);
                }
            }
        }
        return nums;
    }
};

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int n = nums.length;
        int evenindex = 0;
        int oddindex =nums.length-1;
        
        int[] sortedArray = new int[n];
        for(int i=0;i<n;i++){
            if(nums[i]%2 == 0 ){
                sortedArray[evenindex]= nums[i];
                evenindex++;
            }else{
                sortedArray[oddindex]= nums[i];
                oddindex--;
            }
        }
        return sortedArray;
    }
}